src/dmd/cond.d

```diff
@@ -362,13 +373,48 @@ extern (C++) final class StaticForeach : RootObject
         auto catass = new CatAssignExp(aloc, new IdentifierExp(aloc, idres), res[1]);
         s2.push(createForeach(aloc, pparams[1], new ExpStatement(aloc, catass)));
         s2.push(new ReturnStatement(aloc, new IdentifierExp(aloc, idres)));
-        auto aggr = wrapAndCall(aloc, new CompoundStatement(aloc, s2));
-        sc = sc.startCTFE();
-        aggr = aggr.expressionSemantic(sc);
-        aggr = resolveProperties(sc, aggr);
-        sc = sc.endCTFE();
-        aggr = aggr.optimize(WANTvalue);
-        aggr = aggr.ctfeInterpret();
+
+        Expression aggr;
+        Type indexty;
+
+        if (rangefe && (indexty = ety.typeSemantic(aloc, sc)).isintegral())
+        {
+            const lwr = rangefe.lwr.toInteger();
+            auto  upr = rangefe.upr.toInteger();
+            size_t length;
+
+            if (upr < lwr)
+                length = 0;
+            else if (upr - lwr > size_t.max)
+                error(aloc, "`static %s` iterations cannot exceed `size_t.max`, range expression is `(%llu .. %llu)`", Token.toChars(rangefe.op), lwr, upr);
```

Comment:
What about here, I don't think this needs to be size_t, and it makes the compiler have different limits based on how it was compiled. If internally `Expressions/IntegerExp` use the same Array implementation as above, I would just set the limit to `uint.max` instead. Nobody should be making a compile time array that big, I doubt any reasonable person even has a system with enough memory to do that.

Reply:
Yes, it's unlikely a CT array of that size and the compiler will fail long before any large number with out of memory error.
But why show `uint` if in theory the reason is `size_t`. The compiler already has the `size_t` limit on its internal data structures (`Array` is used everywhere).